src/subdomains/supporting/dex/strategies/check-liquidity/impl/defichain-default.strategy.ts

```diff
@@ -67,19 +67,25 @@ export class DeFiChainDefaultStrategy extends CheckLiquidityStrategy {
     const { targetAmount, availableAmount, maxPurchasableAmount, isSlippageDetected, slippageMessage, feeAmount } =
       liquidity;
 
+    const targetAvailableAmount = availableAmount > 0 ? availableAmount : 0;
+    const targetMaxPurchasableAmount = maxPurchasableAmount > 0 ? maxPurchasableAmount : 0;
```

Comment:
Could use `const targetAvailableAmount = Math.max(availableAmount, 0);` if you like 😉  (everywhere)

Reply:
done!